bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,485 @@
+     * Sending REFRESH to any item updates all items, since all values are returned in the response from Smhi.
+     * Therefore there's a wait of 5 seconds before the values are fetched, in which time all other commands are
+     * blocked, to prevent spamming Smhi's API.
+     *
+     * @param channelUID
+     * @param command
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateNow();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SmhiConfiguration.class);
+
+        connection = new SmhiConnector(httpClient);
+
+        // Check which channel groups are selected in the config.
+        List<Channel> channels = new ArrayList<>();
+        channels.addAll(createChannels());
+        updateThing(editThing().withChannels(channels).build());
+
+        startPolling();
+        updateNow();
+    }
+
+    /**
+     * Start polling for updated weather forecast.
+     */
+    private synchronized void startPolling() {
+        logger.debug("Start polling");
+        forecastUpdater = scheduler.scheduleWithFixedDelay(this::waitForForecast, 1, 1, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Cancels all jobs.
+     */
+    private synchronized void cancelPolling() {
+        logger.debug("Cancelling polling");
+        Future<?> localRef = forecastUpdater;
+        if (localRef != null) {
+            localRef.cancel(false);
+        }
+        localRef = instantUpdate;
+        if (localRef != null) {
+            localRef.cancel(false);
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        if (!isModifyingCurrentConfig(configurationParameters)) {
+            return;
+        }
+
+        validateConfigurationParameters(configurationParameters);
+
+        Configuration configuration = editConfiguration();
+        for (Map.Entry<String, Object> configurationParameter : configurationParameters.entrySet()) {
+            configuration.put(configurationParameter.getKey(), configurationParameter.getValue());
+        }
+
+        if (isInitialized()) {
+            updateConfiguration(configuration);
+            config = configuration.as(SmhiConfiguration.class);
+            updateThing(editThing().withChannels(createChannels()).build());
+            if (!isPollingStarted()) {
+                startPolling();
+            }
+            updateNow();
+        } else {
+            // persist new configuration and notify Thing Manager
+            updateConfiguration(configuration);
+            ThingHandlerCallback callback = getCallback();
+            if (callback != null) {
+                callback.configurationUpdated(this.getThing());
+            } else {
+                logger.warn("Handler {} tried updating its configuration although the handler was already disposed.",
```

Comment:
This warning sounds very familiar does the framework issue the same warning?

Reply:
The code in the method is in part copied from BaseThingHandler, I wanted most of the same functionality but skip the dispose()/initialize() part and handle that differently. Is there a better way to achieve this?

Reply:
Why do you need to override the method it seems that you still perform most of the re-initialization anyway?

Reply:
You're right, i'll remove the method and just let it be handled in the default way.